### ISSUE TITLE
Querying by label without full scan

### ISSUE THREAD

**Issue by asaf** [thinkaurelius/titan#776]
Support querying by label without full scan

`g.query('label', 'foo')` causes full scan while the vertex label is there,
What block us from retrieving all vertices by the label without full scan?

The workaround is to create a vertex and attach to it all items but IMO this approach conflicts with the concept of labels in the first place.

More info here:
https://groups.google.com/forum/#!searchin/aureliusgraphs/asaf/aureliusgraphs/f2HX9hUvBuA/7miijge-x5kJ

**Comment by mbroecheler** [comment]
Labels are not indexed since that will likely lead to an oversized index
entry. Hence the full scan.

On Tue, Oct 7, 2014 at 11:33 AM, Asaf Shakarchi notifications@github.com
wrote:

> Support querying by label without full scan
> 
> g.query('label', 'foo') causes full scan while the vertex label is there,
> What block us from retrieving all vertices by the label without full scan?
> 
> The workaround is to create a vertex and attach to it all items but IMO
> this approach conflicts with the concept of labels in the first place.
> 
> More info here:
> 
> https://groups.google.com/forum/#!searchin/aureliusgraphs/asaf/aureliusgraphs/f2HX9hUvBuA/7miijge-x5kJ
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/thinkaurelius/titan/issues/776.

## 

Matthias Broecheler
http://www.matthiasb.com

**Comment by asaf** [comment]
@mbroecheler The vertex that represents a label also have no edges to the vertices of its type?

If not, the only solution to avoid full scan is to create a "root" vertex and attach it to all vertices of its type?

Thanks

**Comment by mbroecheler** [comment]
Yes, you currently would have to build an index or a root vertex to get all
the instances. We are considering a feature that allows you to build a
label specific index.

On Wed, Oct 8, 2014 at 2:40 AM, Asaf Shakarchi notifications@github.com
wrote:

> @mbroecheler https://github.com/mbroecheler The vertex that represents
> a label also have no edges to the vertices of its type?
> 
> If not, the only solution to avoid full scan is to create a "root" vertex
> and attach it to all vertices of its type?
> 
> Thanks
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/thinkaurelius/titan/issues/776#issuecomment-58333434.

## 

Matthias Broecheler
http://www.matthiasb.com

**Comment by asaf** [comment]
@mbroecheler Thanks, it's up to you if to keep this open or not for the 'label specific feature' reference.
